sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+/**
+ * Runtime configuration interface for Spark. To access this, use `SparkSession.conf`.
+ *
+ * @since 2.0.0
+ */
+abstract class RuntimeConfig {
+
+  /**
+   * Sets the given Spark runtime configuration property.
+   *
+   * @since 2.0.0
+   */
+  def set(key: String, value: String): RuntimeConfig
+
+  /**
+   * Sets the given Spark runtime configuration property.
+   *
+   * @since 2.0.0
+   */
+  def set(key: String, value: Boolean): RuntimeConfig
+
+  /**
+   * Sets the given Spark runtime configuration property.
+   *
+   * @since 2.0.0
+   */
+  def set(key: String, value: Long): RuntimeConfig
```

Comment:
Because it is a lot more annoying to have to type

.set("key", "1.0")

than just

.set("key", 1.0)

On Thursday, February 25, 2016, Wenchen Fan <notifications@github.com>
wrote:

> In sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala
> <https://github.com/apache/spark/pull/11378#discussion_r54211858>:
>
> > +   */
> > +  def set(key: String, value: String): RuntimeConfig
> > +
> > +  /**
> > +   * Sets the given Spark runtime configuration property.
> > +   *
> > +   * @since 2.0.0
> > +   */
> > +  def set(key: String, value: Boolean): RuntimeConfig
> > +
> > +  /**
> > +   * Sets the given Spark runtime configuration property.
> > +   *
> > +   * @since 2.0.0
> > +   */
> > +  def set(key: String, value: Long): RuntimeConfig
>
> We don't have getBoolean and getLong, which means we don't really store
> config values as Boolean or Long. So why not we implement these 2 set in
> the base class, i.e. turn Boolean or Long to string and call the abstract
> set?The get returns String anyway.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/11378/files#r54211858>.
>